bind/utils.go

```diff
@@ -192,19 +193,82 @@ func getGoVersion(version string) (int64, int64, error) {
 	return major, minor, nil
 }
 
+var (
+	rxValidPythonName = regexp.MustCompile(`^[\pL_][\pL_\pN]+$`)
+)
+
 func extractPythonName(gname, gdoc string) (string, string, error) {
-	const PythonName = "\ngopy:name "
-	i := strings.Index(gdoc, PythonName)
+	const (
+		PythonName   = "gopy:name "
+		NLPythonName = "\n" + PythonName
+	)
+	i := -1
+	var tag string
+	// Check for either a doc string that starts with our tag,
+	// or as the first token of a newline
+	if strings.HasPrefix(gdoc, PythonName) {
+		i = 0
+		tag = PythonName
+	} else {
+		i = strings.Index(gdoc, NLPythonName)
+		tag = NLPythonName
+	}
 	if i < 0 {
 		return gname, gdoc, nil
 	}
-	s := gdoc[i+len(PythonName):]
+	s := gdoc[i+len(tag):]
 	if end := strings.Index(s, "\n"); end > 0 {
-		validIdPattern := regexp.MustCompile(`^[\pL_][\pL_\pN]+$`)
-		if !validIdPattern.MatchString(s[:end]) {
+		if !isValidPythonName(s[:end]) {
 			return "", "", fmt.Errorf("gopy: invalid identifier: %s", s[:end])
 		}
 		return s[:end], gdoc[:i] + s[end:], nil
 	}
 	return gname, gdoc, nil
 }
+
+// extractPythonNameFieldTag parses a struct field tag and returns
+// a new python name. If the tag is not defined then the original
+// name is returned.
+// If the tag name is specified but is an invalid python identifier,
+// then an error is returned.
+func extractPythonNameFieldTag(gname, tag string) (string, error) {
+	const tagKey = "gopy"
+	if tag == "" {
+		return gname, nil
+	}
+	tags, err := structtag.Parse(tag)
```

Comment:
bringing this whole new dependency seems a bit overkill for something that could -I think- be achieved with just:
```go
v, ok := reflect.StructTag(tag).Lookup(tagKey)
```
couldn't it?

Reply:
Oops. I should have paid closer attention to what was already available in the `reflect` package. Fixed!

Reply:
no worries :)